### ISSUE TITLE
Use pre-downloaded weights in containerised implementation

### ISSUE THREAD

**Issue by tomaroberts** [wasserth/TotalSegmentator#60]
I've got TotalSegmentator running in a Docker container, however, with every run of the container it downloads the model weights which is about 10 minutes of dead time, effectively.

What's the easiest way of telling TotalSegmentator to use weights which exist locally rather than downloading?

Thanks in advance.

**Comment by wasserth** [comment]
The following could be a solution: Mount the directory with the weights into the docker container using the -v option. Inside of the docker container use the environment variable "TOTALSEG_WEIGHTS_PATH" to point to the directory where the weights are mounted to. Then TotalSegmentator will not download the weights but use the existing ones.

**Comment by denbonte** [comment]
Hey @tomaroberts!

Alternatively to @wasserth's (very clever) suggestion, if you don't want to mount an additional volume with the model weights stored inside (e.g., under `/home/user/.totalsegmentator`), you could also pull the weights from Zenodo directly inside the container and store them at the proper location. This might be easier for you (as you won't need to pull the model weights locally).

If you want to take a look at how we did it as part of a repository we're developing to containerize various DL models, you're welcome to do so [here](https://github.com/AIM-Harvard/mhub/blob/main/mhub/totalsegmentator/dockerfiles/cuda12.0/Dockerfile) 🙃  

For now, we're pulling only the weights for the `--fast` mode - but this can be easily changed by copying/pasting code and using [the Zenodo links provided by the dev(s) in this script](https://github.com/wasserth/TotalSegmentator/blob/378d48936f58f40f9f2494b080fcdd064e2497f9/totalsegmentator/libs.py#L90).

**Comment by tomaroberts** [comment]
@denbonte

Thanks, I'll check it out!

**Comment by tomaroberts** [comment]
I have got this working now, thanks @denbonte and @wasserth 

I used a combination of both your suggestions here: https://github.com/GSTT-CSC/TotalSegmentator-AIDE/blob/cdfb930004252b5cd12886bedfdd86c06591270c/app/Dockerfile#L32

Basically, download the weights inside the Docker container, then set the ENV variable `TOTALSEG_WEIGHTS_PATH` inside the container (line 80). This is picked up by TotalSegmentator on execution.

Worked a charm! :)
